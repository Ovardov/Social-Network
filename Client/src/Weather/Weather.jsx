import React from 'react';
import Search from '../Search/Search';
import styles from './weather.module.scss';

function Weather() {
    return (
        <div className={styles.container}>
            <section className={styles.search}>
                <Search />
            </section>

            <section className={styles.information}>
                <p className={styles.icon}>
                    <i className="far fa-sun"></i>
                    {/* Sunny */}
                </p>
                <p className={styles.temperature}>20 &#8451;</p>
                <p className={styles.date}>Monday, 18 November 2019</p>
                <p className={styles.location}>
                    <i className="fas fa-map-marker-alt"></i>
                    Los Angeles
                </p>
            </section>
        </div>
    )
}

export default Weather;