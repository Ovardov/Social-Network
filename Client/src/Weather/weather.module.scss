.container {
    grid-column: 1;
    background: var(--color-blue-main) url('./weather.svg') bottom no-repeat;
    position: relative;
    display: flex;
    flex-direction: column;
    align-items: center;
    text-align: center;
    border-radius: 0.3em;
    padding: 1em;
    color: #FFFFFF;
    overflow: hidden;

    & .search {
        margin-bottom: 1em;

        & form {
            border: 1px solid #FFFFFF;
            border-radius: 0.3em;

            & input[type="text"] {
                border: 1px solid #FFFFFF;
            }
        }
    }

    & .information {
        z-index: 2;

        & .temperature,
        & .icon i {
            font-size: 2.2rem;
        }

        & .temperature {
            padding-bottom: 1em;
        }

        & .icon {
            display: flex;
            flex-direction: column;
            font-size: 1.5em;
            padding-bottom: 0;
        }

        & .date {
            padding-bottom: 0.1rem;
        }

        & .location {
            padding-bottom: 0;

            & i {
                margin-right: 0.3em;
            }
        }
    }
}