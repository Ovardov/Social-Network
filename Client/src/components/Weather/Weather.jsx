import React, { useState, useEffect } from 'react';
import Search from '../Search/Search';
import weatherService from '../../services/weatherService';
import styles from './weather.module.scss';

function Weather() {
    const [city, setCity] = useState('Sofia');
    const [temperature, setTemperature] = useState(0);
    const [weatherDescription, setWeatherDescription] = useState('');
    const [weatherIcon, setIcon] = useState(null);

    useEffect(() => {
        weatherService.loadWeather(city)
            .then(res => {
                setWeatherDescription(res.weather[0].main)
                setTemperature(Math.round(res.main.temp - 273.15));
                setIcon(res.weather[0].icon);
            })
            .catch(err => {
                console.log(err);
            })
    }, [city]);

    const getCity = (e) => {
        e.preventDefault();

        const data = new FormData(e.target);

        setCity(data.get('search'))
    }

    const getDate = () => {
        let date = new Date();

        let day = date.getDate();
        let month = date.getMonth();
        let year = date.getFullYear();

        return `${day}/${month}/${year}`;
    }

    return (
        <div className={styles.container}>
            <section className={styles.search}>
                <Search submit={getCity} />
            </section>

            <section className={styles.information}>
                <p className={styles.temperature}>{temperature} &#8451;</p>
                <p className={styles.icon}>
                    {weatherIcon && <img src={`http://openweathermap.org/img/wn/${weatherIcon}.png`} alt={weatherDescription} />}
                    {weatherDescription}
                </p>
                <p className={styles.location}>
                    <i className="fas fa-map-marker-alt"></i>
                    {city}
                </p>
                <p className={styles.date}>{getDate()}</p>
            </section>
        </div>
    )
}

export default Weather;