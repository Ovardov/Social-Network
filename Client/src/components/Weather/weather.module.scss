.container {
    background: var(--color-blue-main) url('./weather.svg') bottom no-repeat;
    position: relative;
    display: flex;
    flex-direction: column;
    align-items: center;
    text-align: center;
    border-radius: 0.3em;
    padding: 1em;
    color: #FFFFFF;
    overflow: hidden;

    & .search {
        margin-bottom: 1em;

        & form {
            border: 1px solid #FFFFFF;
            border-radius: 0.3em;

            & input[type="text"] {
                border: 1px solid #FFFFFF;
            }
        }
    }

    & .information {
        z-index: 2;

        & .icon,
        & .location,
        & .temperature {
            font-size: 1.3em;
            padding-bottom: 0;
        }

        & .temperature {
            font-size: 2.2rem;
        }

        & .icon {
            display: flex;
            align-items: center;
            padding-bottom: 1.5em;
        }

        & .date {
            padding-bottom: 0.1rem;
        }

        & .location>i {
            margin-right: 0.3em;
        }
    }
}