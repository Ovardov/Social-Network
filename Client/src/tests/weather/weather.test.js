import React from 'react'
import renderer from 'react-test-renderer';
import Weather from '../../components/Weather/Weather';

jest.mock("react-router-dom", () => ({
    Link: "Link"
}));

describe('Weather component', () => {
    test('should render weather for Sofia', () => {
        const component = renderer.create(
            <Weather />
        );

        expect(component.toJSON()).toMatchSnapshot();
    });
})
