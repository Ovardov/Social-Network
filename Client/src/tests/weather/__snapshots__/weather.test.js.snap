// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather component should render weather for Sofia 1`] = `
<div
  className="container"
>
  <section
    className="search"
  >
    <form
      className="container"
      onSubmit={[Function]}
    >
      <p>
        <input
          name="search"
          placeholder="Search"
          type="text"
        />
      </p>
      <button
        type="submit"
      >
        <i
          className="fas fa-search"
        />
      </button>
    </form>
  </section>
  <section
    className="information"
  >
    <p
      className="temperature"
    >
      0
       â„ƒ
    </p>
    <p
      className="icon"
    >
      
    </p>
    <p
      className="location"
    >
      <i
        className="fas fa-map-marker-alt"
      />
      Sofia
    </p>
    <p
      className="date"
    >
      15/11/2019
    </p>
  </section>
</div>
`;
